### DISCUSSION TITLE
USDT/USDC transactions

### DISCUSSION THREAD

**Discussion by nikitaNotFound** [WalletConnect/walletconnect-monorepo#2382] [Developer Support] [answered]
Hey everyone!

Perhaps someone knows how can I make USDT/USDC transactions using WalletConnect?
I learnt from similar question that smart contract address is needed and I should pass into sendTransaction parameters. But should I request for chain permissions first?... So, maybe someone already made it and may share an exact example with me please.

**Comment by glitch-txs** [comment]
Which version and deps of WalletConnect are you using?

**Reply by nikitaNotFound** [reply]
> Which version and deps of WalletConnect are you using?

WC v2 sir

**Reply by glitch-txs** [reply]
Web3modal with wagmi? Or SignClient?

**Reply by nikitaNotFound** [reply]
> Web3modal with wagmi? Or SignClient?

Standalone Web3Modal with SignClient

**Reply by glitch-txs** [reply, marked as answer]
You need to encode data for it, alternatively you can use a library that do that for use, like ethers.js, web3.js or viem. For this you would like to use ethereum-provider

https://docs.walletconnect.com/2.0/web/providers/ethereum

**Reply by nikitaNotFound** [reply]
> You need to encode data for it, alternatively you can use a library that do that for use, like ethers.js, web3.js or viem. For this you would like to use ethereum-provider
> 
> https://docs.walletconnect.com/2.0/web/providers/ethereum

It's not clear for me how the requested chains should look like. For example, that's how it looks for eth, sol and bnb
```
const namespaces = {
    eip155: {
        methods: ['eth_sendTransaction'],
        chains: ['eip155:1', 'eip155:56'],
        events: ['accountsChanged']
    },
    solana: {
        methods: ['eth_sendTransaction'],
        chains: ['solana:8E9rvCKLFQia2Y35HXjjpWzj8weVo44K'],
        events: ['accountsChanged']
    }
};
```

Should I add something here to support USDT?

And the second part is a request. I suppose I should pass a USDT' smart contract address to it, but I don't see how in the documentation.

```
const result = await signClient.request({
    topic: session.topic,
    request: {
        method: 'eth_sendTransaction',
        params: [transaction]
    },
    chainId: 'eip155:1'
});
```

**Comment by finessevanes** [comment]
>Should I add something here to support USDT?

No you don't need to add anything for USDT. Here are the docs from `eth_transaction`: https://ethereum.org/en/developers/docs/apis/json-rpc/#eth_sendtransaction

**Comment by Venheruk10** [comment]
Thanks!
